macaroonbakery/httpbakery/client.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2017 Canonical Ltd.
+# Licensed under the LGPLv3, see LICENCE file for details.
+
+import base64
+try:
+    from cookielib import Cookie, CookieJar         # Python 2
+except ImportError:
+    from http.cookiejar import Cookie, CookieJar    # Python 3  # NOQA
+import requests
+try:
+    from urlparse import urlparse                   # Python 2
+    from urlparse import urljoin                    # Python 2
+except ImportError:
+    from urllib.parse import urlparse               # Python 3  # NOQA
+    from urllib.parse import urljoin                # Python 3  # NOQA
+
+from macaroonbakery.bakery import discharge_all
+from macaroonbakery import utils
+
+ERR_INTERACTION_REQUIRED = 'interaction required'
+ERR_DISCHARGE_REQUIRED = 'macaroon discharge required'
+TIME_OUT = 30
+MAX_DISCHARGE_RETRIES = 3
+
+
+class BakeryAuth:
+    ''' BakeryAuth holds the context for making HTTP requests
+        that automatically acquire and discharge macaroons around the requests
+        framework.
+        Usage:
+            from macaroonbakery import httpbakery
+            jar = requests.cookies.RequestsCookieJar()
+            resp = requests.get('some protected url',
+                                auth=httpbakery.BakeryAuth(cookies=jar))
+            resp.raise_for_status()
+    '''
+    def __init__(self, visit_page=None, key=None,
+                 cookies=requests.cookies.RequestsCookieJar()):
```

Comment:
The usage will be:
                resp = requests.get('some url',
                                    cookies=jar,
                                    auth=httpbakery.BakeryAuth(cookies=jar))
If someone do not provide cookies, the default is to create one that will be valid for the time of the request so we can use the same interface.
                resp = requests.get('some url', auth=httpbakery.BakeryAuth())